pkg/apis/core/v1/defaults.go

```diff
@@ -173,6 +173,29 @@ func SetDefaults_Pod(obj *v1.Pod) {
 			}
 		}
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+		// Set ResizePolicy to default values, if not specified.
+		for i := range obj.Spec.Containers {
+			resources := make(map[v1.ResourceName]bool)
+			for _, p := range obj.Spec.Containers[i].ResizePolicy {
```

Comment:
I'm confused - don't you want to loop over the defined resources in the pod, rather than the policies?  This loop doesn't make sense to me.

Reply:
This loop sets the policy to default value (RestartNotRequired) if user does not specify it. It is orthogonal to resources.

Reply:
We are better off without this default. My code already handles defaults correctly in kubelet if it were not specified. This default has the ugly side-effect of setting it up for BE pods to which resize is not a valid operations. I'll toss this out.

Reply:
I misread it before - it's `!found` rather than `found`.  Up to you how to handle it but if you keep it, a comment would help.